src/Microsoft.VisualStudio.Extensibility.Testing.Xunit.Shared/Harness/VisualStudioInstanceKey.cs

```diff
@@ -53,22 +60,54 @@ public override int GetHashCode()
             hashCode = (hashCode * -1521134295) + Version.GetHashCode();
             hashCode = (hashCode * -1521134295) + EqualityComparer<string>.Default.GetHashCode(RootSuffix);
             hashCode = (hashCode * -1521134295) + MaxAttempts.GetHashCode();
+            hashCode = (hashCode * -1521134295) + EnvironmentVariables.Count.GetHashCode();
             return hashCode;
         }
 
         public string SerializeToString()
         {
-            return $"{(int)Version};{RootSuffix};{MaxAttempts}";
+            var builder = new StringBuilder();
+            builder.Append(EnvironmentVariables.Count);
+            foreach (var environmentVariable in EnvironmentVariables)
+            {
+                builder.Append(';');
+                builder.Append(Convert.ToBase64String(Encoding.UTF8.GetBytes(environmentVariable)));
+            }
+
+            return $"{(int)Version};{RootSuffix};{MaxAttempts};{builder}";
         }
 
         public static VisualStudioInstanceKey DeserializeFromString(string s)
         {
             var semicolon1 = s.IndexOf(';');
             var semicolon2 = s.IndexOf(';', semicolon1 + 1);
+            var semicolon3 = s.IndexOf(';', semicolon2 + 1);
+            var semicolon4 = s.IndexOf(';', semicolon3 + 1);
+            var environmentVariables = new string[0];
```

Comment:
Avoid creating empty arrays. Use `Array.Empty<string>()` instead.

Reply:
`Array.Empty<string>` isn't available on all TFMs supported by this library.